### PR TITLE
Fix ASMOSESUTest  for failing simple_regression problem

### PR DESCRIPTION
**Pull request by Eman22S**

### CODE REVIEW

File: opencog/atomese/interpreter/Interpreter.cc [hunk 1]
```diff
@@ -129,8 +129,12 @@ ValuePtr Interpreter::execute(const Type t, const ValueSeq &params)
 		LinkValuePtr result(new LinkValue(_result));
 
 		for (const ValuePtr &p : params) {
+			std::string test2 = oc_to_string(p);
+			int m=0;
```
Comment (ngeiswei) [inline opencog/atomese/interpreter/Interpreter.cc]: @Eman22S I believe these 2 lines are cruft from some past debugging.

File: opencog/atomese/interpreter/Interpreter.cc [hunk 2]
```diff
@@ -129,8 +129,12 @@ ValuePtr Interpreter::execute(const Type t, const ValueSeq &params)
...
 			result = logical_and(result, LinkValueCast(p));
 		}
+		std::string test3 = oc_to_string(ValuePtr(result));
+		int m=0;
```
Comment (ngeiswei) [inline opencog/atomese/interpreter/Interpreter.cc]: Same remark about cruft.

File: opencog/combo/type_checker/type_tree.h [hunk 3]
```diff
@@ -259,6 +259,8 @@ void infer_arg_type_tree(const combo_tree& tr, type_tree_seq& arg_types);
 // insert the types of the arguments in arg_types in the signature of
 // a given function type. If the function is a constant then wrap it
 // in lambda_type
+type_node_seq type_tree_to_tyn_seq(const type_tree& tt);
```
Comment (ngeiswei) [inline opencog/combo/type_checker/type_tree.h]: @Eman22S could you move below the comment above `type_tree_to_tyn_seq` as it corresponds to `insert_arg_type_tree`, and add a comment explaining what `type_tree_to_tyn_seq` does?

**Comment by ngeiswei** [comment]
I think that looks good @Eman22S, thanks. @kasimebrahim, does that look good to you too?

**Comment by kasimebrahim** [comment]
Is there a reason why you didn't make size optional in the Interpreter? If there is a reason for it then checking if the _problem_data_size is not given is unnecessary and niether is extract_output_size.
https://github.com/singnet/asmoses/blob/55919832d9c7cd5e330239ac231216e5a8a00441/opencog/atomese/interpreter/Interpreter.cc#L54-L55

**Comment by Eman22S** [comment]
@kasimebrahim Yes, we can sure remove `extract_output_size` (I must  have forgotten to remove it) because it is not computing the output_size correctly i.e it had some implementation flaws. its better we pass the output_size directly .

**Comment by ngeiswei** [comment]
@kasimebrahim, can I leave it to you to merge, if you determine it is ready? Thanks.

**Comment by ngeiswei** [comment]
@kasimebrahim @Eman22S do you remember if that one is ready for merging?

**Pull request closed without merging**